src/charm.py

```diff
@@ -28,142 +42,101 @@ def __init__(self, msg: str, status_type=None):
         self.status = status_type(self.msg)
 
 
-class Operator(CharmBase):
+class KubeflowDashboardOperator(CharmBase):
+    """A Juju Charm for Kubeflow Dashboard Operator"""
+
     def __init__(self, *args):
         super().__init__(*args)
 
-        self.log = logging.getLogger(__name__)
-        self.image = OCIImageResource(self, "oci-image")
-
+        self.logger = logging.getLogger(__name__)
+        self._namespace = self.model.name
+        self._lightkube_field_manager = "lightkube"
+        self._profiles_service = None
+        self._name = self.model.app.name
+        self._service = "npm start"
+        self._container_name = "kubeflow-dashboard"
+        self._container = self.unit.get_container(self._name)
```

Comment:
I am, I think unjustifiably, a little little worried that `self.unit.get_container(self._name)` might return a copy of this object rather than a pointer to the object.  If that is the case, we'd effectively be making a copy here and then whenever we use it, it could be stale.  I don't see any obvious signs that this will be a copy in the code, but I didn't see any contract saying it would be a live reference either.  

Since this is used only in the `self.container` property, what about refactoring to make that property:
```
@property
def container(self):
  return self.unit.get_container(self._name)
```

Yes, the charm will have to invoke the `get_container` func a few extra times, but this way you don't depend on how they're passing the data back.

Also, if you know the return either can never change and is always a live reference, then ignore this.  Ignore this too if this is a common pattern I just don't know

Reply:
We use this patter in training/katib/dex (all sidecar charms). But it does not mean its good. But I see similar patter in Jons kubernets dashboard charm (although he calls it multiple times in the code instead of assigning it to class param). I vote for tech topic :). But its a valid worry I like the thought process.

Reply:
Good plan.  I'll make a tech topic, can you add links to any charms I miss?  thx!